sklearn/utils/estimator_checks.py

```diff
@@ -305,9 +355,38 @@ def check_estimator(Estimator):
     estimator : estimator object or class
         Estimator to check. Estimator is a class object or instance.
 
+    generate_only : bool, optional (default=False)
+        When `False`, checks are evaluated when `check_estimator` is called.
+        When `True`, `check_estimator` returns a list of checks for the
+        `estimator`.
+
+        .. versionadded:: 0.22
+
+    Notes
+    -----
+    Setting `generate_only=False` allows for checks to be disaseembled when
+    testing an estimator. To use this with pytest this can be done with:
+
+    .. code-block:: python
+
+        from itertools import chain
+        import pytest
+        from sklearn.utils.estimator_checks import check_estimator
+        from sklearn.utils.estimator_checks import set_check_estimator_ids
+
+        @pytest.mark.parametrize(
```

Comment:
Should we be defining a helper that constructs this parameterization?

Reply:
maybe? I wouldn't worry about it for this PR though. It would create more public API surface while we're still figuring things out.

Reply:
The point of a helper would be to get around the obscurity of using chain etc. Yes, it's not needed in this pr, but it looks like @NicolasHug finds the current example surprisingly unreadable